libraries/tock-register-interface/src/fields.rs

```diff
@@ -479,6 +480,112 @@ macro_rules! register_bitmasks {
     };
 }
 
+/// Helper macro for defining debugging of a field.
+#[macro_export]
+#[cfg(not(feature = "register_debug"))]
```

Comment:
If we're not generating any code that is used "by default" (e.g., when not calling `.debug()`), I don't think we need to gate this behind a feature flag.

Reply:
The macro uses `debug::RegisterDebugInfo` so we can't remove it, as using it without the flag would result in `unresolved reference`

Reply:
But another way we can do it, is to have the `debug` module available at all times, just disable the `debug()` functions

Reply:
I have added this change, now its there for all features, but can't be used normally.